src/dmd/utils.d

```diff
@@ -210,6 +140,79 @@ void escapePath(OutBuffer* buf, const(char)* fname)
     }
 }
 
+/**
+ * Takes a path, and make it compatible with GNU Makefile format.
+ *
+ * GNU make uses a weird quoting scheme for white space.
+ * A space or tab preceded by 2N+1 backslashes represents N backslashes followed by space;
+ * a space or tab preceded by 2N backslashes represents N backslashes at the end of a file name;
+ * and backslashes in other contexts should not be doubled.
+ *
+ * Params:
+ *   buf = Buffer to write the escaped path to
+ *   fname = Path to escape
+ */
+void escapeMakePath(OutBuffer *buf, const(char)* fname)
```

Comment:
Since most function that output to an `Outbuffer` start with `writeX`, I suggest renaming this function for better clarity. (Also the first time I read the usage of the function, I though it was returning a new string, instead of writing the escaped result to a buffer.)

```suggestion
void writeMakePath(OutBuffer *buf, const(char)[] filename)
```

Additionally, accept a D string filename, instead of a C string, as it makes for cleaner (as safer) usage:

```D
buf.writeMakePath(params.exefile);
```

vs

```D
escapeMakePath(buf, &params.exefile[0]);
```